src/model.jl

```diff
@@ -579,10 +601,26 @@ function _build_model(
     # Parameters
     bus_demand = _make_bus_demand(case, start_index, end_index) * demand_scaling
     # Demand flexibility parameters (if present)
+    bus_demand_flex_cost_up = nothing
+    bus_demand_flex_cost_dn = nothing
     if demand_flexibility.enabled
         bus_demand_flex_amt = _make_bus_demand_flexibility_amount(
             case, demand_flexibility, start_index, end_index
         )
+        if !isnothing(demand_flexibility.cost_up)
+            bus_demand_flex_cost_up = permutedims(
+                Matrix(
+                    demand_flexibility.cost_up[start_index:end_index, 2:end]
+                )
+            )
+        end
+        if !isnothing(demand_flexibility.cost_dn)
+            bus_demand_flex_cost_dn = permutedims(
+                Matrix(
+                    demand_flexibility.cost_dn[start_index:end_index, 2:end]
+                )
+            )
+        end
```

Comment:
Can we move these blocks within the `_add_objective_function!` function? The reason that we calculate `bus_demand_flex_amt` within `_build_model` is because we need it to set bounds for the `load_shift_up` and `load_shift_up` variables, but I believe we can continue to pass only `demand_flexibility` to `_add_objective_function!` and handle the rest within there.

Reply:
Agreed. That would also prevent `bus_demand_flex_cost_up` and `bus_demand_flex_cost_dn` from needing to be initialized as `nothing` a few lines above.

Reply:
@danielolsen @lanesmith Hi, I just saw the comments, I will address them and update this PR soon.